src/parser/parser.cpp

```diff
@@ -665,10 +665,10 @@ api::Term Parser::nextExpression()
 {
   Debug("parser") << "nextExpression()" << std::endl;
   d_resourceManager->spendResource(ResourceManager::Resource::ParseStep);
-  Expr result;
+  api::Term result;
   if (!done()) {
     try {
-      result = d_input->parseExpr().getExpr();
+      result = d_input->parseExpr();
```

Comment:
The `return api::Term(result);` statement below can be replaced by just `return result;`, I think.